### PR TITLE
feat: add sso enforcement for invite signup

### PR DESCRIPTION
**Pull request by zlwaterfield**
## Problem

Right now, SSO enforcement is not upheld on the invite signup page. An invited user can login with password or one of our social SSO options (Google, Github, Gitlab) and there is no SAML option.

Here is what it looks like:
Screenshot 2024-10-24 at 5 30 12 PM

If the domain of the invite has SSO enforcement on then we want to show the correct button (SAML, Google, etc.) like we do on the login page. And we need to make sure we pass the invite ID through so it can accept the invite as well.

## Changes

This PR adds the precheck call to check on the invite signup page using the domain of the email being invited. If it's turned on it then will hide all other fields (password, name, etc.) as well as the social SSO buttons along the bottom and show the SSO login button instead. 

It also adds an `actionText` to the SSO login button props so we can show "Continue with.." instead of "Login with".

Note: with this change we will no login take in the `Role` field. We could look into adding it into the support fields for the SSO extra fields so it's processed on the SSO callback. 

Here is what is would look like:
Screenshot 2024-10-24 at 5 27 43 PM

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

## Does this work well for both Cloud and self-hosted?

Yes

## How did you test this code?

Need to do some more manual testing on this to confirm the invite gets accepted correctly. Here is where it happens: https://github.com/PostHog/posthog/blob/master/posthog/api/signup.py#L517-L519

### CODE REVIEW

**Comment by raquelmsmith** [comment]
I haven't reviewed the code yet but saw there are no changes to the backend, there should prob be some validation there?

**Comment by zlwaterfield** [comment]
> I haven't reviewed the code yet but saw there are no changes to the backend, there should prob be some validation there?

What sorta validation would you expect? I haven't changed any forms so it's using all of the same logic. The only thing is pushing to SSO if it's enforced.

**Comment by zlwaterfield** [comment]
> @zlwaterfield You shouldn't be able to use the API to sign up with username/password when SSO is enforced.

Ah yes, I can add that in.

**Comment by zlwaterfield** [comment]
@raquelmsmith added with tests.

**Comment by zlwaterfield** [comment]
Looking into why snapshots changed because they shouldn't have.

File: frontend/src/scenes/authentication/InviteSignup.tsx [hunk 1]
```diff
@@ -221,49 +232,73 @@ function UnauthenticatedAcceptInvite({ invite }: { invite: PrevalidatedInvite })
...
-                    status="alt"
-                    htmlType="submit"
-                    data-attr="password-signup"
-                    loading={isSignupSubmitting}
-                    center
-                    fullWidth
-                >
-                    Continue
-                </LemonButton>
+                {precheckResponse.status === 'pending' || !precheckResponse.sso_enforcement ? (
```
Comment (raquelmsmith) [inline frontend/src/scenes/authentication/InviteSignup.tsx:273]: Are these conditionals correct? If the response status is pending do we want to show the "continue" button?

File: frontend/src/scenes/authentication/InviteSignup.tsx [hunk 2]
```diff
@@ -221,49 +232,73 @@ function UnauthenticatedAcceptInvite({ invite }: { invite: PrevalidatedInvite })
...
+                    </LemonButton>
+                ) : (
+                    <SSOEnforcedLoginButton
+                        provider={precheckResponse.sso_enforcement}
+                        email={login.email}
+                        actionText="Continue"
+                        extraQueryParams={invite ? { invite_id: invite.id } : undefined}
+                    />
+                )}
+                {precheckResponse.saml_available && !precheckResponse.sso_enforcement && (
```
Comment (raquelmsmith) [inline frontend/src/scenes/authentication/InviteSignup.tsx:294]: And is this conditional correct? If we **don't** have `precheckResponse.sso_enforcement` we will show the `SSOEnforcedLoginButton` with `provider="saml"`?

**Review by raquelmsmith — APPROVED** [review]
Just verifying the conditionals are correct, otherwise looks good 👍

File: frontend/src/scenes/authentication/InviteSignup.tsx (hunk 1 above)
Comment (zlwaterfield) [reply frontend/src/scenes/authentication/InviteSignup.tsx:273]: Will test a few things and confirm / improve.

File: frontend/src/scenes/authentication/InviteSignup.tsx (hunk 2 above)
Comment (zlwaterfield) [reply frontend/src/scenes/authentication/InviteSignup.tsx:300]: @raquelmsmith, it was correct, but I agree that the logic was hard to follow. Can you give it one final review?

**Review by raquelmsmith — APPROVED** [review]
